components/eventing-controller/api/v1alpha2/constants.go

```diff
@@ -0,0 +1,23 @@
+package v1alpha2
+
+const (
+	STANDARD TypeMatching = "standard"
+	EXACT    TypeMatching = "exact"
```

Comment:
I propose to use a prefix like `TypeMatching`. This makes it easier for the different values to be discovered.

```suggestion
	TypeMatchingStandard TypeMatching = "standard"
	TypeMatchingExact     TypeMatching  = "exact"
```

<details>
  <summary>There are other examples (also in the golang stdlib) which follow this idea:</summary>

[source](https://www.sohamkamani.com/golang/enums/)

**net/http package**
- https://pkg.go.dev/net/http#pkg-constants
```go
const (
	MethodGet     = "GET"
	MethodHead    = "HEAD"
	MethodPost    = "POST"
	MethodPut     = "PUT"
	MethodPatch   = "PATCH" // [RFC 5789](https://rfc-editor.org/rfc/rfc5789.html)
	MethodDelete  = "DELETE"
	MethodConnect = "CONNECT"
	MethodOptions = "OPTIONS"
	MethodTrace   = "TRACE"
)
```

**Color package**
- https://pkg.go.dev/image/color#pkg-variables
This is an example where no prefix is required since the package name `color` is already meaningful enough. Example: `color.Black`

```go
var (
	Black       = [Gray16](https://pkg.go.dev/image/color#Gray16){0}
	White       = [Gray16](https://pkg.go.dev/image/color#Gray16){0xffff}
	Transparent = [Alpha16](https://pkg.go.dev/image/color#Alpha16){0}
	Opaque      = [Alpha16](https://pkg.go.dev/image/color#Alpha16){0xffff}
)
```

**Log package**:
```go
const (
	Ldate         = 1 << [iota](https://pkg.go.dev/builtin#iota)     // the date in the local time zone: 2009/01/23
	Ltime                         // the time in the local time zone: 01:23:23
	Lmicroseconds                 // microsecond resolution: 01:23:23.123123.  assumes Ltime.
	Llongfile                     // full file name and line number: /a/b/c/d.go:23
	Lshortfile                    // final file name element and line number: d.go:23. overrides Llongfile
	LUTC                          // if Ldate or Ltime is set, use UTC rather than the local time zone
	Lmsgprefix                    // move the "prefix" from the beginning of the line to before the message
	LstdFlags     = [Ldate](https://pkg.go.dev/log@go1.16.6#Ldate) | [Ltime](https://pkg.go.dev/log@go1.16.6#Ltime) // initial values for the standard logger
)
```

</details>